### ISSUE TITLE
Wrapper for C_WaitForSlotEvent

### ISSUE THREAD

**Issue by arjennienhuis** [parallaxsecond/rust-cryptoki#145]
I might need to use C_WaitForSlotEvent. I wrote this and it seems to work:

```rust
    fn wait_for_slot_event_impl(&self, dont_block: bool) -> Result<Slot> {
        let flags = if dont_block { CKF_DONT_BLOCK } else { 0 };
        unsafe {
            let mut slot: CK_SLOT_ID = 0;
            let wait_for_slot_event = get_pkcs11!(self, C_WaitForSlotEvent);
            let rv = wait_for_slot_event(flags, &mut slot, std::ptr::null_mut());
            Rv::from(rv).into_result()?;
            Ok(Slot::new(slot))
        }
    }

    /// wait for slot events (insertion or removal of a token)
    pub fn wait_for_slot_event(&self) -> Result<Slot> {
        self.wait_for_slot_event_impl(false)
    }

    /// get the latest slot event (insertion or removal of a token)
    pub fn get_slot_event(&self) -> Result<Option<Slot>> {
        match self.wait_for_slot_event_impl(true) {
            Err(Error::Pkcs11(RvError::NoEvent)) => Ok(None),
            Ok(slot) => Ok(Some(slot)),
            Err(x) => Err(x),
        }
    }

```

Is that something I should make a pull request for?

**Comment by ionut-arm** [comment]
Hey!

Yes, please! That looks pretty good, and if you could also add a test or two that would be much appreciated 🙏🏻

**Comment by wiktor-k** [comment]
Tests for that may be a little bit tricky since they'd require waiting for the event in one thread and then making SoftHSM trigger the event in another... but definitely it's better to have tests than not :sweat_smile:

**Comment by ionut-arm** [comment]
As far as I could tell from the spec, you can trigger an event before and this function should pick it up. But even if you need to trigger it during the test, it shouldn't be too difficult to implement. Don't want to put too much pressure, though, so happy either way!

**Comment by arjennienhuis** [comment]
Implemented by #146
